packages/hardhat-core/src/internal/util/io-ts.ts

```diff
@@ -1,5 +1,28 @@
 import * as t from "io-ts";
 
+export function optionalOrNullable<TypeT, OutputT, InputT>(
+  codec: t.Type<TypeT, OutputT, InputT>,
+  name: string = `${codec.name} | undefined`
+): t.Type<TypeT | undefined, OutputT | undefined, InputT | undefined | null> {
+  return new t.Type(
+    name,
+    (u: unknown): u is TypeT | undefined =>
+      u === undefined || u === null || codec.is(u),
```

Comment:
I'm still confused about the difference between the type guard and the decoding function, but I think this shouldn't have the `|| u === null` part. Also notice that it's not compatible with the signature, which is saying that `u is TypeT | undefined`

Reply:
Yes, `=== null` shouldn't be here.

What makes the type guard vs decoding function so weird is that TypeT can be more general than OutputT. What for? What's the use case for this? I have no idea.

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1524/commits/6aad332ce28da8583cf291cab7a7e32ee12f9425